### PR TITLE
Replace custom clang-format bits with stock meson targets

### PR DESCRIPTION
**Pull request by cgwalters**
This is built into meson:
https://mesonbuild.com/Code-formatting.html

### CODE REVIEW

**Comment by cgwalters** [comment]
Fun; I tried to update to ubuntu-24.04 across the board but it seems the fuse test starts erroring out with that...will defer that tech debt to later...

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -6,6 +6,20 @@ permissions:
   actions: read
 
 jobs:
+  clang-format:
+    # To pick up a newer clang
+    runs-on: ubuntu-24.04
```
Comment (allisonkarlitskaya) [inline .github/workflows/test.yaml:11]: Any reason not to move the main workflow over to 24.04?  We're probably gonna get exposed to that in the next couple of months anyway (wieh it becomes the new `ubuntu-latest`)... makes sense to shake out any issues ahead of time.

Comment (cgwalters) [reply .github/workflows/test.yaml:11]: I added a comment about that https://github.com/containers/composefs/pull/326#issuecomment-2321626211

Comment (allisonkarlitskaya) [reply .github/workflows/test.yaml:11]: Sorry.  Missed that.

**Review by allisonkarlitskaya — APPROVED** [review]

**Pull request merged**